lib/DevServerManagerV2.ts

```diff
@@ -0,0 +1,139 @@
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import { i18n } from './lang';
+import { promptUser } from './prompts/promptUtils';
+import { DevModeInterface as UIEDevModeInterface } from '@hubspot/ui-extensions-dev-server';
+import {
+  startPortManagerServer,
+  stopPortManagerServer,
+  requestPorts,
+} from '@hubspot/local-dev-lib/portManager';
+import {
+  getHubSpotApiOrigin,
+  getHubSpotWebsiteOrigin,
+} from '@hubspot/local-dev-lib/urls';
+import { getAccountConfig } from '@hubspot/local-dev-lib/config';
+import { ProjectConfig } from '../types/Projects';
+import { IntermediateRepresentationNodeLocalDev } from '@hubspot/project-parsing-lib/src/lib/types';
+
+const i18nKey = 'lib.DevServerManager';
+
+type DevServerInterface = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+  setup?: Function;
+  start?: (options: object) => Promise<void>;
+  fileChange?: (filePath: string, event: string) => Promise<void>;
+  cleanup?: () => Promise<void>;
+};
+
+class DevServerManagerV2 {
+  private initialized: boolean;
+  private started: boolean;
+  private devServers: DevServerInterface[];
+
+  constructor() {
+    this.initialized = false;
+    this.started = false;
+    this.devServers = [UIEDevModeInterface];
+  }
+
+  async iterateDevServers(
```

Comment:
An explanatory note would be helpful with this function. Is this handling if the user has multiple dev servers running?

Reply:
See comment below, but this is just repeated functionality from the existing `DevServerManager`. Basically, its set up to handle the case where there are multiple dev servers (even though there's only one right now) and call each lifecycle method on each of them

Reply:
I forget how exactly this manifests itself, but I'm pretty sure the way I initially built this causes errors thrown by the dev servers to get swallowed. No need to fix it in this PR, but we may want to eventually revisit it.